src/DETHRACE/common/structur.c

```diff
@@ -128,14 +129,46 @@ void RaceCompleted(tRace_over_reason pReason) {
 // IDA: void __usercall Checkpoint(int pCheckpoint_index@<EAX>, int pDo_sound@<EDX>)
 void Checkpoint(int pCheckpoint_index, int pDo_sound) {
     LOG_TRACE("(%d, %d)", pCheckpoint_index, pDo_sound);
-    NOT_IMPLEMENTED();
+
+    PratcamEvent(33);  // FIXME: or PratcamEventNow
+    DoFancyHeadup(12);
+    if (pDo_sound) {
+        DRS3StartSound(gIndexed_outlets[4], 8012);
+    }
 }
 
 // IDA: void __cdecl IncrementCheckpoint()
 void IncrementCheckpoint() {
     int done_voice;
     LOG_TRACE("()");
-    NOT_IMPLEMENTED();
+
+    done_voice = 0;
+    if (gRace_finished) {
+        return;
+    }
+    gLast_checkpoint_time = GetTotalTime();
+    if (gCheckpoint < gCheckpoint_count) {
+        gCheckpoint++;
+    } else {
+        gCheckpoint = 1;
+        gLap++;
+        if (gLap == gTotal_laps) {
+            PratcamEvent(33);  // FIXME: or PratcamEventNow
```

Comment:
What does the comment mean about PratcamEventNow?

Reply:
I'm not sure this function is PratcamEvent or PratcamEventNow.
It's probably PratcamEvent, but hence the comment

Reply:
I agree, its `PratcamEvent`. The DOS version is the most closely related to the symbol dump for ordering of functions and globals, confirmed with that